code/__HELPERS/unsorted.dm

```diff
@@ -457,36 +457,35 @@ Turf and target are separate in case you want to teleport some distance from a t
 
 /atom/proc/GetAllContents(var/T)
 	var/list/processing_list = list(src)
-	var/list/assembled = list()
 	if(T)
-		while(processing_list.len)
-			var/atom/A = processing_list[1]
-			processing_list.Cut(1, 2)
+		. = list()
+		var/i = 0
+		while(i < length(processing_list))
+			var/atom/A = processing_list[++i]
 			//Byond does not allow things to be in multiple contents, or double parent-child hierarchies, so only += is needed
 			//This is also why we don't need to check against assembled as we go along
 			processing_list += A.contents
 			if(istype(A,T))
-				assembled += A
+				. += A
 	else
-		while(processing_list.len)
-			var/atom/A = processing_list[1]
-			processing_list.Cut(1, 2)
+		var/i = 0
+		while(i < length(processing_list))
+			var/atom/A = processing_list[++i]
 			processing_list += A.contents
-			assembled += A
-	return assembled
+		return processing_list
```

Comment:
```suggestion
		return processing_list - src
```
Otherwise it will return src too.

Reply:
Disregard, this behaviour existed before and it might be intended even though it seems illogical.